### ISSUE TITLE
Prune invalid tickets function bug

### ISSUE THREAD

**Issue by CreatorX8** [vertyco/vrt-cogs#122]
**Is your feature request related to a problem? Please describe.**
Prune invalid tickets function has a bug. When the max number of tickets is set to one and only one user open a ticket and the channel is deleted mannualy. When is called this function the entry in the database stays and the user cannot open more tickets (because max is set to one). There is no problem if there is at least one valid ticket which belongs to another user. When there is one valid and one invalid ticket and the function is called there is no problem.

**Describe the solution you'd like**
To fix this issue you have to remove one line of code in tickets/utils.py
Line 281: **if valid_user_tickets:** has to be removed
![image](https://github.com/vertyco/vrt-cogs/assets/90335627/bc767c2c-54d3-481c-b1f5-7b0f42d785b8)

This fix the error and we don't need this if statement beacause we have valid_user_tickets = {} and it is filled if there is any valid ticket for the user. If there is no valid tickets it will set valid_opened_tickets[user_id] = {} because the initial value of valid_user_tickets is {}!

Sincerely,
Creator X

**Comment by vertyco** [comment]
While your proposed solution may not break anything, i do not believe it will not fix the issue you are having. 
Here is the code that checks how many tickets a user has opened:
```python
opened = conf["opened"]
user_can_close = conf["user_can_close"]
logchannel = guild.get_channel(panel["log_channel"]) if panel["log_channel"] else None
uid = str(user.id)
if uid in opened and max_tickets <= len(opened[uid]):
    em = discord.Embed(
        description=_("You have the maximum amount of tickets opened already!"),
        color=discord.Color.red(),
    )
    return await interaction.response.send_message(embed=em, ephemeral=True)
```

With the current way the tickets are pruned, the user's ID would be cleared entirely from the config, thus it wouldn't show up and be caught by the ticket count when trying to open a ticket.

**Comment by CreatorX8** [comment]
The problem is that when the limit is one ticket and only one user in the server created a ticket and then manually delete his channel and execute the command **[p]tickets cleanup** it is not cleared in the database because of this check.

**Comment by vertyco** [comment]
Found the issue, it should only be checking if count is higher than 0. just pushed the fix, ty for reporting
Old code:
```python
    if valid_opened_tickets and count:
        await config.guild(guild).opened.set(valid_opened_tickets)
```

New code:
```python
    if count:
        await config.guild(guild).opened.set(valid_opened_tickets)
```

**Comment by CreatorX8** [comment]
Yes, and this is a solution. When I first noticed the problem the first thing I did was this, but then I changed the other check because it worked that way too.

**Comment by vertyco** [comment]
your suggested fix only worked coincidentally, we don't want to store and empty dict for a user but rather clean it up if they have no tickets.

**Comment by CreatorX8** [comment]
Ok, you are absolutely right but then, when you close all your tickets we have and empty dict for the user and it is not cleaned up.

**Comment by vertyco** [comment]
That's true, i'll add an extra bit to clean that up
